smsdk/utils.py

```diff
@@ -26,4 +26,18 @@ def get_url(protocol, tenant, site_domain):
     :type site_domain: :class:`string`
     """
 
-    return "{}://{}.{}".format(protocol, tenant, site_domain)
\ No newline at end of file
+    return "{}://{}.{}".format(protocol, tenant, site_domain)
+
+def check_kw(kw) -> bool:
+    for key in [
+        "machine_type", 
+        "Machine",
+        "machine__source", 
+        "End Time",
+        "endtime",
+        "Start Time",
+        "starttime"
+    ]:
+        if key in kw or kw[0] == "_":
```

Comment:
Start with the single char check as that is faster.  Should this check be case insensitive?  `key.lower()`? can key be empty string or something other than string?

Reply:
included suggestioon, key is list of str and in lower case already